datashader/datatypes.py

```diff
@@ -645,9 +647,10 @@ def tolist(self):
         else:
             return list(self)
 
-    def __array__(self, dtype=None):
+    def __array__(self, dtype=None, copy=None):
         dtype = np.dtype(object) if dtype is None else np.dtype(dtype)
-        return np.asarray(self.tolist(), dtype=dtype)
+        arr = np.asarray(self.tolist(), dtype=dtype)
+        return arr.copy() if copy else arr
```

Comment:
As above, this will always create a copy anyway.

```suggestion
        return np.array(self.tolist(), dtype=dtype)
```